.ci/run_e2e_tests.sh

```diff
@@ -16,7 +25,6 @@ if [[ -z ${DOCKER_USERNAME} ]]; then
     echo "Skipping Byzantine Tests because Docker Credentials were not present."
     BYZANTINE_IMAGE=""
 else
-    echo "$DOCKER_PASS" | docker login --username "$DOCKER_USERNAME" --password-stdin
```

Comment:
What I don't understand though is that even here there was the login *before* the pull.

So how does your PR change the situation we were having?

Reply:
Yeah I think this whole CI could do with a fresh review 🤔 

I moved the `login` to before the `bash "${SRC_PATH}"/scripts/build_image.sh` which pulls the `FROM golang:1.15.5-buster`

Reply:
aaaah gotcha, that's where we were pulling before authenticating. :+1: